File: internal/models/mock_content.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// MockContent represents mock content for a URL in the system.
type MockContent struct {
	ID          uint           `gorm:"primarykey" json:"id"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`
	URLID       uint           `gorm:"not null" json:"url_id"`
	Url         *Url           `gorm:"foreignKey:URLID" json:"url,omitempty"` // Changed from URL to Url to match struct name
	Randomness  int            `json:"randomness"`
	Latency     int64          `json:"latency"`
	Description string         `json:"description"`
	Name        string         `gorm:"not null" json:"name"`
	Data        string         `gorm:"type:text" json:"data"`
}

// TableName specifies the table name for the MockContent model.
func (MockContent) TableName() string {
	return "mock_content"
}

File: src/main/java/com/mock_json/mock_api/models/MockContent.java
package com.mock_json.mock_api.models;

import jakarta.persistence.*;
import lombok.*;

import org.hibernate.annotations.SoftDelete;
import org.hibernate.annotations.UpdateTimestamp;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;

import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "mock_content")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@SoftDelete
public class MockContent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JsonBackReference
    @JoinColumn(name = "url_id", nullable = false)
    private Url urlId;

    @Builder.Default
    @Column(nullable = false)
    private Long randomness = 0L;

    @Builder.Default
    @Column(nullable = false)
    private Long latency = 0L;

    @Column(nullable = true)
    private String description;

    @Column(nullable = false)
    private String name;

    @Lob
    @JsonProperty("data")
    @Column(nullable = false, columnDefinition = "LONGTEXT")
    private String data;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    @JsonIgnore
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    @JsonIgnore
    private LocalDateTime updatedAt;

}
